playbooks/roles/streisand-gateway/templates/vhost.j2

```diff
@@ -9,23 +9,43 @@ server {
   ssl_certificate      {{ nginx_self_signed_certificate }};
   ssl_certificate_key  {{ nginx_private_key }};
 {% endif %}
-  add_header Strict-Transport-Security max-age=631138519;
 
   auth_basic            "Authorization Required";
   auth_basic_user_file  {{ streisand_gateway_htpasswd_file }};
 
+  ssl_protocols TLSv1.2;
+  ssl_prefer_server_ciphers on;
   ssl_dhparam {{ dhparams_location }};
+  ssl_ciphers ECDHE-RSA-AES256-GCM-SHA512:DHE-RSA-AES256-GCM-SHA512:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:DHE-RSA-AES256-SHA;
+  ssl_ecdh_curve secp384r1;
+  ssl_session_timeout  10m;
+  ssl_session_cache shared:SSL:10m;
+  ssl_session_tickets off;
+  ssl_stapling on;
```

Comment:
I don't think there's much value in setting `ssl_stapling` and `ssl_stapling_verify` for a self-signed PKI.  Can you scope this addition to the mode in which a Let's Encrypt certificate has been issued?